test/adapter/json_api/pagination_links_test.rb

```diff
@@ -0,0 +1,114 @@
+require 'test_helper'
+require 'will_paginate/array'
+require 'kaminari'
+require 'kaminari/hooks'
+::Kaminari::Hooks.init
+
+module ActiveModel
+  class Serializer
+    class Adapter
+      class JsonApi
+        class PaginationLinksTest < Minitest::Test
+          URI = 'http://example.com'
+
+          def setup
+            ActionController::Base.cache_store.clear
+            @array = [
+              Profile.new({ id: 1, name: 'Name 1', description: 'Description 1', comments: 'Comments 1' }),
+              Profile.new({ id: 2, name: 'Name 2', description: 'Description 2', comments: 'Comments 2' }),
+              Profile.new({ id: 3, name: 'Name 3', description: 'Description 3', comments: 'Comments 3' })
+            ]
+          end
+
+          def mock_request(query_parameters={}, original_url=URI)
+            context = Minitest::Mock.new
+            context.expect(:original_url, original_url )
+            context.expect(:query_parameters, query_parameters)
+            @options = {}
+            @options[:context] = context
+          end
+
+          def using_kaminari
+            Kaminari.paginate_array(@array).page(2).per(1)
+          end
+
+          def using_will_paginate
+            @array.paginate(page: 2, per_page: 1)
+          end
+
+          def data
+            { data:[
+                { id:"1", type:"profiles", attributes:{name:"Name 1", description:"Description 1" } },
+                { id:"2", type:"profiles", attributes:{name:"Name 2", description:"Description 2" } },
+                { id:"3", type:"profiles", attributes:{name:"Name 3", description:"Description 3" } }
+              ]
+            }
+          end
+
+          def links
+            {
+              links:{
+                self: "#{URI}?page%5Bnumber%5D=2&page%5Bsize%5D=1",
+                first: "#{URI}?page%5Bnumber%5D=1&page%5Bsize%5D=1",
+                prev: "#{URI}?page%5Bnumber%5D=1&page%5Bsize%5D=1",
+                next: "#{URI}?page%5Bnumber%5D=3&page%5Bsize%5D=1",
+                last: "#{URI}?page%5Bnumber%5D=3&page%5Bsize%5D=1"
+              }
+            }
+          end
+
+          def expected_response_without_pagination_links
+            data
+          end
+
+          def expected_response_with_pagination_links
+            {}.tap do |hash|
+              hash[:data] = [data.values.flatten.second]
+              hash.merge! links
+            end
```

Comment:
overall, I'm happy with this PR and I think it's ok to merge.  I'd like to change this to something like

```ruby
expected_response_without_pagination_links.merge(links)
```